File: src/dataset/c/kindergarten_garden_Exercism_m3g4d1v3r/claude/ClaudeSonnet4_kindergarten_garden_v4.c
#include "kindergarten_garden.h"

#include <stdio.h>

static const char *students[] = {"Alice",  "Bob",    "Charlie", "David",
                                 "Eve",    "Fred",   "Ginny",   "Harriet",
                                 "Ileana", "Joseph", "Kincaid", "Larry"};

static const plant_t plant_lookup[256] = {
    ['C'] = CLOVER,
    ['G'] = GRASS,
    ['R'] = RADISHES,
    ['V'] = VIOLETS
};

size_t rowlen(const char *diagram) {
    const char *start = diagram;
    const char *end = diagram;
    const size_t max_len = ARRAY_SIZE(students) * 2 + 1;
    
    while (*end != '\n' && (size_t)(end - start) < max_len) {
        end++;
    }
    
    return (*end == '\n') ? (size_t)(end - start) : 0;
}

plants_t plants(const char *diagram, const char *student) {
    plants_t result = {{DIRT, DIRT, DIRT, DIRT}};
    
    if (!diagram || !student) return result;
    
    const size_t diag_rowlen = rowlen(diagram);
    if (diag_rowlen == 0) return result;
    
    int student_idx = -1;
    for (size_t idx = 0; idx < ARRAY_SIZE(students); idx++) {
        if (strcmp(student, students[idx]) == 0) {
            student_idx = (int)idx;
            break;
        }
    }
    
    if (student_idx == -1) return result;
    
    const size_t base_pos = 2 * (size_t)student_idx;
    if (base_pos + 1 >= diag_rowlen) return result;
    
    const size_t second_row_pos = base_pos + diag_rowlen + 1;
    const size_t diag_len = strlen(diagram);
    if (second_row_pos + 1 >= diag_len) return result;
    
    result.plants[0] = plant_lookup[(unsigned char)diagram[base_pos]];
    result.plants[1] = plant_lookup[(unsigned char)diagram[base_pos + 1]];
    result.plants[2] = plant_lookup[(unsigned char)diagram[second_row_pos]];
    result.plants[3] = plant_lookup[(unsigned char)diagram[second_row_pos + 1]];
    
    return result;
}

File: src/dataset/typescript/kindergarten-garden_Exercism-typescript-FilipeCerejo/claude/ClaudeSonnet4_kindergarten-garden_v4.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Garden = void 0;
const DEFAULT_STUDENTS = [
    'Alice',
    'Bob',
    'Charlie',
    'David',
    'Eve',
    'Fred',
    'Ginny',
    'Harriet',
    'Ileana',
    'Joseph',
    'Kincaid',
    'Larry',
];
const PLANT_CODES = {
    G: 'grass',
    V: 'violets',
    R: 'radishes',
    C: 'clover',
};
class Garden {
    constructor(diagram, students = DEFAULT_STUDENTS) {
        this._pots = diagram.split('\n').map((line) => line.split(''));
        const sortedStudents = [...students].sort();
        this._studentIndexMap = new Map();
        for (let i = 0; i < sortedStudents.length; i++) {
            this._studentIndexMap.set(sortedStudents[i], i);
        }
    }
    plants(student) {
        const studentIndex = this._studentIndexMap.get(student);
        if (studentIndex === undefined)
            return [];
        const firstPlantIdx = studentIndex * 2;
        const row0 = this._pots[0];
        const row1 = this._pots[1];
        return [
            PLANT_CODES[row0[firstPlantIdx]],
            PLANT_CODES[row0[firstPlantIdx + 1]],
            PLANT_CODES[row1[firstPlantIdx]],
            PLANT_CODES[row1[firstPlantIdx + 1]],
        ];
    }
}
exports.Garden = Garden;
